src/java.base/share/classes/java/util/zip/InflaterInputStream.java

```diff
@@ -252,6 +252,10 @@ public void close() throws IOException {
     /**
      * Fills input buffer with more data to decompress.
      * @throws    IOException if an I/O error has occurred
+     * @throws    EOFException if the end of input stream has been reached
+     *            unexpectedly
+     * @apiNote
+     * The {@link #len} field will be updated when this method is invoked.
```

Comment:
I don't think an apiNote will work here as it need to be normative text. I suspect this issue will require specifying fill to read up to buf.length bytes into buf starting at element 0 and setting len to the number of bytes.

Reply:
> I don't think an apiNote will work here as it need to be normative text. I suspect this issue will require specifying fill to read up to buf.length bytes into buf starting at element 0 and setting len to the number of bytes.

Revised per your suggestion